editHack.js

```diff
@@ -0,0 +1,49 @@
+const mongoose = require("mongoose");
+
+let conn = null;
+const url = `mongodb://${process.env.MONGO_USER}:${process.env.MONGO_PASSWORD}@ds157136.mlab.com:57136/hackone`;
+
+export const edit = async (event, context) => {
+  context.callbackWaitsForEmptyEventLoop = false;
+  const data = JSON.parse(event.body);
+  const { goal, title, description, hackId } = data;
+  let update = {};
+  if (conn == null) {
+    conn = await mongoose.createConnection(url, {
+      bufferCommands: false, // Disable mongoose buffering
+      bufferMaxEntries: 0 // and MongoDB driver buffering
+    });
+    conn.model(
+      "Hack",
+      new mongoose.Schema({ title: String, description: String, goal: String })
+    );
+  }
+  const Hack = conn.model("Hack");
+
+  // guard against empty fields
+  if (goal) update.goal = goal;
+  if (title) update.title = title;
+  if (description) update.description = description;
+
+  try {
+    const result = await Hack.findByIdAndUpdate(hackId, update, { new: true });
```

Comment:
if you did an upsert, would you be able to use the same lambda for adding and editing a hack?

Reply:
Can research upsert, originally wanted to reuse addHack but decided to keep it lean

Reply:
Upsert looks like a good option with minimum logic involved. Will test it out tomorrow